src/wp-includes/taxonomy.php

```diff
@@ -719,6 +721,16 @@ function get_taxonomy_labels( $tax ) {
 
 	$labels = _get_custom_object_labels( $tax, $nohier_vs_hier_defaults );
 
+	if ( ! isset( $tax->labels->template_name ) && isset( $tax->object_type[0] ) ) {
+		$post_type = get_post_type_object( $tax->object_type[0] );
+
+		if ( isset( $post_type->labels->singular_name ) ) {
+			$post_type_singular_name = $post_type->labels->singular_name;
+			/* translators: %s: Post type name. */
+			$labels->template_name = sprintf( __( '%s Archives' ), $post_type_singular_name );
+		}
+	}
```

Comment:
```suggestion
	if ( ! isset( $tax->labels->template_name ) && ! empty( $tax->object_type ) ) {
		if ( isset( $tax->labels->singular_name ) ) {
			/* translators: %s: Taxonomy name. */
			$labels->template_name = sprintf( _x( '%s Archives', 'taxonomy template name' ), $tax->labels->singular_name );
		}
	}
```

It was showing the post types name rather than the taxonomy's name

Reply:
Yeah, good point. In my original commit it was intentional to use the post type singular name instead of the taxonomy's name, as some plugins have a very generic `singular_name` for taxonomies. Ie, in WooCommerce the [_Product categories_ post type singular name](https://github.com/woocommerce/woocommerce/blob/8cbf2356ffb67285389004a9ed179bc383d6b507/plugins/woocommerce/includes/class-wc-post-types.php#L98) is _Category_ and the [_Product tags_ singular name](https://github.com/woocommerce/woocommerce/blob/8cbf2356ffb67285389004a9ed179bc383d6b507/plugins/woocommerce/includes/class-wc-post-types.php#L141) is _Tag_.

But thinking about it twice and seeing how it's currently done in Gutenberg, it makes more sense to use the taxonomy's singular name. Changed in 9da18e4c229d97c14ff832537e3b6eece6f702dd.